packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmAnyImpl.kt

```diff
@@ -127,27 +127,27 @@ internal class RealmAnyImpl<T : Any> constructor(
         if (other.type != this.type) return false
         if (clazz == ByteArray::class) {
             if (other.internalValue !is ByteArray) return false
-            if (!other.internalValue.contentEquals(this.internalValue as ByteArray)) return false
+            return other.internalValue.contentEquals(this.internalValue as ByteArray)
         } else if (internalValue is BsonObjectId) {
```

Comment:
We only support the KBSON variant for RealmAny, right? And not also our deprecated variant?

Reply:
Right. There is only an entry point for `org.mongodb.kbson.BsonObjectId`. I have removed any logic guarding this and relied on type checking and internal value's `equals` method for this.